test/config.toml

```diff
@@ -10,6 +10,11 @@ ConcurrencyLimit = 56
 MaxGasFeeCap = "148 Gwei"
 BalanceAlertThreshold = "2.3 ether"
 
+[Electrs]
```

Comment:
Given the recent discussion about supporting ElectrumX RPC API, I am not sure if this should be named `[Electrs]` or `[Electrum]`. @nkuba thoughts?

Reply:
@nkuba @pdyraga 
Alternatively, we could keep the name as `electrs` and treat it as kind of a placeholder and update it once [Kuba's PR](https://github.com/keep-network/keep-core/pull/3406) is merged and we know what config is used there and which fields are required to be configured.

Reply:
We could introduce a parent `bitcoin` section, so:
```
[bitcoin.electrum]
URL=.....
```

If we decide to keep multiple APIs supported we could add them in te future as in the example below:

```
[bitcoin]
[electrum]
URL=.....
[electrs]
URL=....
```

Reply:
I think `[bitcoin]` section makes a lot of sense, especially since we have `[ethereum]`. I just wonder if we couldn't keep it simple like:
```
[bitcoin]
  ElectrumURL = ""
```

ElectrumURL does not mean it has to be ElectrumX or Electrs. That's just an Electrum API-compatible node.

Reply:
OK, I've just realized there are more options for electrum than just URL. Then I think `[bitcoin.electrum]` is the best one for now.

Reply:
Done in https://github.com/keep-network/keep-core/blob/b26f5297258e43a7f57047a0157e5f3b111adab5/pkg/bitcoin/config/config.go#L8.

Unfortunately, there is some complication: in order to have `bitcoin.electrum` configuration, I had to define [this Config struct](https://github.com/keep-network/keep-core/blob/b26f5297258e43a7f57047a0157e5f3b111adab5/pkg/bitcoin/config/config.go#L8) somwhere.

I couldn't do this in the `bitcoin` package because of circular dependency problem (`electrum` imports `bitcoin` and `bitcoin` would import `electrum`). It would also mean that the specific bitcoin implementation (electrum) leaks into its parent package.

Therefore I decided to create a separate package `pkg/bitcoin/config` that would contain the configs for all the possible bitcoin implementations (electrum, electrs, etc).

I don't like this idea too much. Are you ok with this or should I do sth else?

Reply:
Why not just in `pkg/config`? Package `pkg/bitcoin` does not have to contain it.

Reply:
@lukasz-zimnoch 
Yes, I also thought about it.
Done in 96b0e94e3ce9f1faca845bc3e2a3ee5a26a100b5.

BTW, should it be `keep-core/pkg/config` or simply `keep-core/config`? I chose the first option, but I had to create a new directory for `pkg/config`.

Reply:
`pkg/bitcoin` defines Bitcoin chain abstractions and `pkg/bitcoin/electrum` uses these abstractions. I think this dependency makes sense and we should leave it as-is.

On the other hand, all other configuration structures lie in respective packages now: `libp2p`, `storage`, `clientinfo`, `ethereum`... `pkg/config` with `BitcoinConfig` is the only exception which I am not a huge fan of. 

Wouldn't it be better to move `BitcoinConfig` to `pkg/bitcoin` and embed electrum configuration in this file? It is still not ideal but better in my opinion:

```
type BitcoinConfig struct {
   Electrum ElectrumConfig 
}

type ElectrumConfig struct {
	URL string
}
```

Alternatively, introduce `pkg/bitcoin/config` package. Given we have just one configuration option for electrum now, I'd prefer the first solution as the simplest one.

Reply:
I looked at `pkg/bitcoin` one more time and you are right @lukasz-zimnoch. This package is just about Bitcoin types, as the referenced documentation explains. From this perspective, we are not configuring anything in `pkg/bitcoin` but we are configuring Electrum.

From the config file's perspective though, I think it makes sense to have the `bitcoin` section, especially that we have `ethereum` section with an URL there and we do not care if it is Geth, Nethermind, Erigon, or others.

What about having`[bitcoin.electrum]` in the config file, just like we do now but having the config structure in `pkg/bitcoin/electrum`?

Reply:
> What about having[bitcoin.electrum] in the config file, just like we do now but having the config structure in pkg/bitcoin/electrum?

But what if we ever want to add another typy of Bitcoin implementation (e.g. `electrs`) and use them both, so that we have:
```
[bitcoin]
  [electrum]
    URL = ...
  [electrs]
    URL = ... 
```

We would be defining the Bitcoin structure with `electrum` and `electrs` in electrum?

Reply:
Yeah, this is the best option IMO. Having `bitcoin` section is a pure config thing and it should be declared there.

Reply:
Moved the `BitcoinConfig` struct to the electrum package in 232b141cac2dc47b6aff6b3e24c924a85aeb157a.
Hope this is what you wanted.

Reply:
Yep!